packages/mongodb-chatbot-ui/src/services/conversations.ts

```diff
@@ -222,12 +254,11 @@ export class ConversationService {
     };
 
     await fetchEventSource(this.getUrl(path, { stream: "true" }), {
+      ...this.fetchOptions,
+      // Need to convert Headers to plain object for fetchEventSource
+      headers: Object.fromEntries(this.fetchOptions?.headers ?? new Headers()),
```

Comment:
Shouldn't `this.fetchOptions` and `this.fetchOptions.headers` always be defined since we assign it in the constructor?